text/0000-github-issue-labels.md

```diff
@@ -0,0 +1,123 @@
+
+The aim is to make it easier for new and existing contributors to work on Gatsby without adding too much admin overhead. It's important for the labelling system to be quick to manage to ensure that it's used!
+
+# Detailed design
+
+The broad areas discussed were:
+
+- Type: labels like `bug`, `discussion`, `feature request`
+- Status: lables like `needs repro`, `ready`
+- Misc: labels like `good first issue`
+- Target (area of responsibility): Use GitHub projects
+- Roadmap: Use GitHub Milestones
+
+At a minimum every new issue should have a _Type_ label applied to it, everything else is optional.  Labelling can be fluid - a lot of issues might start with the label 'type: discussion', before being changed to a more specific _type_.
+
+Taking that information along with the existing repo structure gives a setup something like this:
+
+### Type label
+
+Every issue should have exactly one _Type_ label.
+
+labels: `bug`, `discussion`, `feature`, `maintenance`, `documentation`
+
+### Status label
+
+Optional
+
+labels: `needs repro`, `help wanted`
+
+### Misc label
+
+Optional
+
+So far these are labels indicating that an issue is a well-defined piece of work or labels used by bots.
+
+label: `good first issue`, `hacktober`, `stale?`, `review`
+
+### Projects
+
+Optional
+
+Assign an issue to one of the existing [projects](https://github.com/gatsbyjs/gatsby/projects): docs, v2, UX (add projects as required)
+
+> Question: Do we need to create any additional projects?
+
+### Milestones
+
+We talked about using GitHub's Milestones as a way of defining a roadmap however, Gatsby already has a [`v2` Project](https://github.com/gatsbyjs/gatsby/projects/2) which contains more info than a Milestone could. In the interest of keeping this plan as simple as possible, I'm going to drop Milestones for now (unless anyone has strong objections). This could be looked at again once the labelling and projects is in place.
+
+### Existing labels
+
+There are 20 existing labels. While we're adding new labels, it makes sense to revisit the existing ones. Having a small, clearly defined set of labels makes it much more likely that the labelling system will be used. Here's what I propose should happen to each of them:
+
+| Label  | Destiny | Notes |
+|---|---|---|
+| API/Plugins | ❌delete | too specific (currently applied to 66 issues) |
+| bug | ✅keep |  rename to `type: bug` or similar |
+| Documentation | ❌delete | redundant - add all issues with this label to the 'Documentation' project before deleting the label |
+| DX | ❌delete |  too specific (currently applied to 13 issues) |
+| Feature Request | ✅keep | rename to `type: feature` or similar |
+| good first issue | ✅keep | exact name is used by GitHub, don't rename |
+| GraphQL | ❌delete | too specific (currently applied to 13 issues) |
+| Hacktoberfest | ✅keep | exact name is used by Hacktoberfest, don't rename  |
```

Comment:
I wonder if there's a way to automatically add this tag to ones we tag with "good first issue"? Otherwise we'd need to manually keep the two in sync.

Reply:
Yeah I think this would be a good next step. Maybe using the existing probot or something like [Danger JS](http://danger.systems/js/).